Explorer/Assets/DCL/WorldTime/WorldTimeProvider.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Cysharp.Threading.Tasks;
+using DCL.Diagnostics;
+using DCL.Optimization.PerformanceBudgeting;
+using ECS.Prioritization.Components;
+using ECS.StreamableLoading.Common.Components;
+using ECS.StreamableLoading.Common.Systems;
+using System;
+using System.Threading;
+using UnityEngine.Networking;
+
+namespace DCL.Time
+{
+    public class WorldTimeProvider : IWorldTimeProvider
+    {
+        private const string TIME_SERVER_URL = "https://peer.decentraland.org/about";
+        private const int TIME_BETWEEN_UPDATES = 5000;
+        private const float GAME_HOURS_PER_CYCLE = 24;
+        private const float REAL_MINUTES_PER_CYCLE = 120;
+
+        private DateTime cachedServerTime;
+        private DateTime cachedSystemTime;
+        private float cachedTimeInSeconds;
+        private bool isPaused;
+
+        public void SetPausedState(bool isPaused)
+        {
+            this.isPaused = isPaused;
+        }
+
+        public async UniTask<float> GetWorldTimeAsync(CancellationToken ct)
+        {
+            if (isPaused) return cachedTimeInSeconds;
+
+            DateTime currentSystemTime = DateTime.Now;
+            TimeSpan timeDifference = currentSystemTime - cachedServerTime;
```

Comment:
Do you consider the time zone? It is probably better to use UTC

Reply:
you are right, I was converting server time to UniversalTime but not local time!

Reply:
You could save a conversion by doing `DateTime currentSystemTime = DateTime.UtcNow;`